src/DictVectors/initiators.jl

```diff
@@ -42,302 +78,147 @@ Concrete implementations:
-    len = length(dvec)
-    entries = length(dvec) == 1 ? "entry" : "entries"
-    print(io, "InitiatorDVec{$K,$V} with $len $entries, style = $(dvec.style), initiator = $(dvec.initiator)")
-end
-
-###
-### Interface
-###
-StochasticStyle(dv::InitiatorDVec) = dv.style
-storage(dv::InitiatorDVec) = dv.storage
-
-function Base.getindex(dvec::InitiatorDVec{<:Any,V}, add) where {V}
-    return value(dvec.initiator, get(dvec.storage, add, zero(InitiatorValue{V})))
-end
-function Base.setindex!(dvec::InitiatorDVec{<:Any,V}, v, k) where {V}
-    if iszero(v)
-        delete!(dvec.storage, k)
-    else
-        dvec.storage[k] = InitiatorValue{V}(safe=v)
-    end
-    return v
-end
-
-function Base.get(dvec::InitiatorDVec, args...)
-    return value(dvec.initiator, get(dvec.storage, args...))
-end
-function Base.get!(dvec::InitiatorDVec{<:Any,V}, key, default) where {V}
-    return value(dvec.initiator, get!(dvec.storage, key, InitiatorValue{V}(safe=default)))
-end
-function Base.get!(f::Function, dvec::InitiatorDVec{<:Any,V}, key) where {V}
-    return value(dvec.initiator, get!(dvec.storage, key, InitiatorValue{V}(safe=f())))
-end
-
-@delegate InitiatorDVec.storage [haskey, getkey, pop!, isempty, length, keys]
-@delegate_return_parent InitiatorDVec.storage [delete!, empty!, sizehint!]
 
 """
-    deposit!(w::InitiatorDVec, add, val, p_add=>p_val)
-Add `val` into `w` at address `add` as an [`InitiatorValue`](@ref).
-"""
-function deposit!(w::InitiatorDVec, add, val, (p_add, p_val))
-    V = valtype(w)
-    i = w.initiator
-    old_val = get(w.storage, add, zero(InitiatorValue{V}))
-    if p_add == add # diagonal death
-        if abs(p_val) > i.threshold
-            new_val = InitiatorValue{V}(initiator=val)
-        else
-            new_val = InitiatorValue{V}(safe=val)
-        end
-    else # offdiagonal spawn
-        if abs(p_val) > i.threshold
-            new_val = InitiatorValue{V}(safe=val)
-        else
-            new_val = InitiatorValue{V}(unsafe=val)
-        end
-    end
-    new_val += old_val
-    if new_val == InitiatorValue{V}(0, 0, 0)
-        delete!(w.storage, add)
-    else
-        w.storage[add] = new_val
-    end
-    return w
-end
+    NoInitiator{V} <: InitiatorRule{V}
 
-function deposit!(w::InitiatorDVec{<:Any,V}, add, val::InitiatorValue{V}, _) where {V}
-    dict = storage(w)
-    old_val = get(dict, add, zero(valtype(dict)))
-    dict[add] = old_val + val
-    return w
-end
-
-###
-### Iterators
-###
-# These are needed because `Iterators.map` does not infer `eltype` correctly.
-"""
-    InitiatorIterator
+Default initiator rule for [`PDVec`](@ref), that disables the approximation.
 
-Iterator over pairs or values of an `InitiatorDVec`.
+See [`InitiatorRule`](@ref).
 """
-struct InitiatorIterator{T,D,I}
-    iter::D
-    initiator::I
-
-    InitiatorIterator{T}(iter::D, initiator::I) where {T,D,I} = new{T,D,I}(iter, initiator)
-end
+struct NoInitiator <: InitiatorRule end
```

Comment:
```suggestion
struct NonInitiator <: InitiatorRule end
```
Use search and replace as this appears in many places!